### PR TITLE
[TG-2585] Support implicitly generic superclasses

### PR DESCRIPTION
**Pull request by mgudemann**
Classes with implicitly generic superclasses were not supported in the creation of Java generic symbol types.

### CODE REVIEW

**Review by peterschrammel — APPROVED** [review]
As mentioned previously, I don't like the manual parsing - something to reconsider if this code gets longer.

**Comment by mgudemann** [comment]
@peterschrammel I fully agree with you, this should become a technical debt ticket

File: unit/java_bytecode/java_bytecode_parse_generics/parse_derived_generic_class.cpp [hunk 1]
```diff
@@ -5,7 +5,7 @@
  Author: DiffBlue Limited. All rights reserved.
 
 \*******************************************************************/
-
+#include <iostream>
```
Comment (majakusber) [inline unit/java_bytecode/java_bytecode_parse_generics/parse_derived_generic_class.cpp]: Leftover debug?

File: unit/java_bytecode/java_bytecode_parse_generics/parse_derived_generic_class.cpp [hunk 2]
```diff
@@ -559,4 +559,69 @@ SCENARIO(
...
+
+    const symbolt &derived_symbol = new_symbol_table.lookup_ref(class_prefix);
+    const class_typet &derived_class_type =
+      require_type::require_java_implicitly_generic_class(derived_symbol.type);
+
+    THEN("The base for superclass is generic *and* implicitly generic")
+    {
+      REQUIRE(derived_class_type.bases().size() == 1);
+      const typet &base_type = derived_class_type.bases().at(0).type();
+      std::cout << "INFO: " << base_type.pretty() << std::endl;
```
Comment (majakusber) [inline unit/java_bytecode/java_bytecode_parse_generics/parse_derived_generic_class.cpp]: Leftover debug

**Review by majakusber — APPROVED** [review]
Looks good, just some debug output left in there. Thank you for extending the tests!

File: unit/java_bytecode/java_bytecode_parse_generics/parse_derived_generic_class.cpp (hunk 2 above)
Comment (mgudemann) [reply unit/java_bytecode/java_bytecode_parse_generics/parse_derived_generic_class.cpp]: @svorenova fixed, thanks!

**Pull request merged**